packages/docmanager/src/dialogs.ts

```diff
@@ -237,24 +238,23 @@ class CreateFromHandler extends Widget {
   /**
    * Show the createNew dialog.
    */
-  showDialog(): Promise<Widget> {
+  showDialog(): Promise<Widget | null> {
     return showDialog({
       title: `Create New ${this._creatorName}`,
       body: this.node,
```

Comment:
this `showDialog` API is pretty weird. `this` is a widget, and passing `this.node` to `showDialog(...)` means that it's not being treated like a widget, and will never get lifecycle messages or be disposed.

If the handler is a widget, it should always be treated as a widget. If it doesn't need to be a widget, then it shouldn't inherit `Widget`.

Maybe we can update the dialog api to just take a vdom node for content?